docs/proposals/autoscaling.md

```diff
@@ -0,0 +1,245 @@
+## Abstract
+Auto-scaling is a data-driven feature that allows users to increase or decrease capacity as needed by controlling the 
+number of pods deployed within the system automatically.  
+
+## Motivation
+
+Applications experience peaks and valleys in usage.  In order to respond to increases and decreases in load, administrators 
+scale their applications by adding computing resources.  In the cloud computing environment this can be 
+done automatically based on statistical analysis and thresholds.
+
+### Goals
+
+* Provide a concrete proposal for implementing auto-scaling pods within Kubernetes
+* Implementation proposal should be in line with current discussions in existing issues: 
+    * Resize verb - [1629](https://github.com/GoogleCloudPlatform/kubernetes/issues/1629)
+    * Config conflicts - [Config](https://github.com/GoogleCloudPlatform/kubernetes/blob/c7cb991987193d4ca33544137a5cb7d0292cf7df/docs/config.md#automated-re-configuration-processes)
+    * Rolling updates - [1353](https://github.com/GoogleCloudPlatform/kubernetes/issues/1353)
+    * Multiple scalable types - [1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)  
+
+## Constraints and Assumptions
+
+* This proposal is for horizontal scaling only.  Vertical scaling will be handled in [issue 2072](https://github.com/GoogleCloudPlatform/kubernetes/issues/2072)
+* `ReplicationControllers` will not know about the auto-scaler, they are the target of the auto-scaler.  The `ReplicationController` responsibilities are 
+constrained to only ensuring that the desired number of pods are operational per the [Replication Controller Design](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/replication-controller.md#responsibilities-of-the-replication-controller)
+* Auto-scalers will be loosely coupled with data gathering components in order to allow a wide variety of input sources
+* Auto-scalable resources will support a resize verb ([1629](https://github.com/GoogleCloudPlatform/kubernetes/issues/1629)) 
+such that the auto-scaler does not directly manipulate the underlying resource.
+* Initially, most thresholds will be set by application administrators. It should be possible for an autoscaler to be 
+written later that sets thresholds automatically based on past behavior (CPU used vs incoming requests).
+* The auto-scaler must be aware of user defined actions so it does not override them unintentionally (for instance someone 
+explicitly setting the replica count to 0 should mean that the auto-scaler does not try to scale the application up)
+* It should be possible to write and deploy a custom auto-scaler without modifying existing auto-scalers
+* Auto-scalers must be able to monitor multiple replication controllers while only targeting a single scalable object
```

Comment:
"Scalable object" is not well-known terminology so I'd suggest being a bit redundant with something you have later, and write here "targeting a single scalable object (for now a ReplicationController, but in the future it could be a job or any resource that implements resize)"